osm-fudge/bk_tree.py

```diff
@@ -0,0 +1,42 @@
+'''An implementation of bk-tree
+
+More: https://en.wikipedia.org/wiki/BK-tree
+'''
+
+class BKTree:
+    '''A bk-tree implementation and defined user methods
+       for insertion and querying the string
+    '''
+
+    def __init__(self, root, distanceMetrics, options):
```

Comment:
Not sure about the root argument here. Initializing a bktree should only take a distance metric, no?

Reply:
Yes, you are right. Right now building a tree is coupled with presence of first word. But it can still exist empty without any words. Gonna remove it and handle the tree initialisation in insert method